WalletWasabi/Helpers/Guard.cs

```diff
@@ -179,5 +182,32 @@ public static string Correct(string? str)
 				? ""
 				: str.Trim();
 		}
+
+		[DebuggerStepThrough]
+		public static GroupElement NotNullOrInfinity(string parameterName, GroupElement groupElement)
+			=> groupElement?.IsInfinity switch
+			{
+				null => throw new ArgumentNullException(parameterName),
+				true => throw new ArgumentException("Point at infinity is not a valid value.", parameterName),
+				false => groupElement
+			};
+
+		[DebuggerStepThrough]
+		public static IEnumerable<GroupElement> NotNullOrInfinity(string parameterName, IEnumerable<GroupElement> groupElements)
+			=> groupElements switch
+			{
+				null => throw new ArgumentNullException(parameterName),
+				_ when !groupElements.Any() => throw new ArgumentException(parameterName),
+				_ => groupElements.Select((ge, i) => NotNullOrInfinity($"{parameterName}[{i}]", ge)).ToList()
+			};
+
+		[DebuggerStepThrough]
+		public static Scalar NotZero(string parameterName, Scalar scalar)
+			=> scalar switch
+			{
+				_ when scalar.IsZero => throw new ArgumentException("Value cannot be zero.", parameterName),
+				_ => scalar
```

Comment:
would not this be a problem when a client wants to create zero-credentials? There, zero-scalars are needed by definition, right?

Reply:
No. We have tests that verify zero-value credentials can be created. This is used only in specific places where zero is "forbidden" but that can happen in a couple of zillon years.